src/utils.sh

```diff
@@ -50,14 +52,23 @@ declare -a EXIT_CODES=(0    # success
 ######################################
 
 function set_opt_out_selection() {
-    if [[ $# == 1 && $1 == true ]];
+    if [ $1 == true ];
     then
         OPT_IN=false
         log_info "The user has opted out of sending usage telemetry."
     else
         OPT_IN=true
         log_info "The user has opted in for sending usage telemetry."
     fi
+
+    # handle correlation vector
+    if [ -z $2 ];
+    then
+        CORRELATION_VECTOR=$(uuidgen)
```

Comment:
uuidgen requires the package 'uuid-runtime' and is not installed on all distributions by default. can we use uuid instead?